js/expiring_messages.js

```diff
@@ -31,8 +31,13 @@
             console.log('next message expires', new Date(expires_at).toISOString());
 
             var wait = expires_at - Date.now();
+
+            // In the past
             if (wait < 0) { wait = 0; }
 
+            // Too far in the future, since it's limited to a 32-bit value
+            if (wait > 2147483647) { wait = 2147483647; }
```

Comment:
iOS is managing our timers differently, so this might not apply, but we check expiring messages at least once every 5 mins when the app is active as a CYA.